Tests/webcomponents/js/ez-universaldiscovery.js

```diff
@@ -0,0 +1,339 @@
+                assertViewRendered(renderViewSpy, window.eZ.YUI.Y);
+            });
+        });
+    });
+
+    describe('UDW init', function () {
+        let renderViewStub;
+        const container = document.createElement('div');
+        const location = getModelMock();
+        const content = getModelMock();
+        const contentType = getModelMock();
+
+        function assertEventParam(e) {
+            assert.equal(e.detail.selection.content, modelObject);
+            assert.equal(e.detail.selection.location, modelObject);
+            assert.equal(e.detail.selection.contentType, modelObject);
+            assert.isFalse(e.bubbles);
+        }
+
+        beforeEach(function () {
+            universalDiscovery = fixture('BasicTestFixture');
+            defineGlobals();
+        });
+
+        afterEach(function () {
+            delete window.eZ;
+            renderViewStub.restore();
+        });
+
+        it('should append the YUI view container', function () {
+            const view = getViewMock(container);
+
+            renderViewStub = sinon.stub(window.eZ.YUI.app, 'renderSideView', function (View, Service, params, done) {
+                assert.equal(
+                    universalDiscovery.getAttribute('title'),
+                    params.title,
+                    "The title should be passed in the parameters"
+                );
+
+                assert.equal(
+                    universalDiscovery.multiple,
+                    params.multiple,
+                    "The multiple should be passed in the parameters"
+                );
+
+                assert.equal(
+                    universalDiscovery.getAttribute('visible-method'),
+                    params.visibleMethod,
+                    "The visibleMethod should be passed in the parameters"
+                );
+
+                assert.equal(
+                    universalDiscovery.getAttribute('confirm-label'),
+                    params.confirmLabel,
+                    "The confirmLabel should be passed in the parameters"
+                );
+
+                assert.equal(
+                    universalDiscovery.getAttribute('starting-location-id'),
+                    params.startingLocationId,
+                    "The startingLocationId should be passed in the parameters"
+                );
+
+                assert.equal(
+                    universalDiscovery.getAttribute('min-discover-depth'),
+                    params.minDiscoverDepth,
+                    "The minDiscoverDepth should be passed in the parameters"
+                );
+
+                done(false, 'whatever', view);
+            });
+            document.dispatchEvent(new CustomEvent('ez:yui-app:ready'));
+            assert.equal(
+                container.parentNode,
+                universalDiscovery
+            );
+        });
+
+        it('should have a cancelDiscoverHandler', function () {
```

Comment:
the 3 following tests try to test too many things at the same time. For each of them, you test both the fact that a `cancelDiscoverHandler`, `contentDiscoveredHandler` or a `isSelectable` function is passed and what those functions are doing.

In a suite called ~~`UDW init`~~ `YUI Universal Discovery Widget`, you should limit this test to check that those function are passed, so the tests should be nameed:

* `should have received a contentDiscoveredHandler function`
* `should have received a cancelDiscoveredHandler function`
* `should have received a isSelectable function`

And in addition to that, there should be a top level `events` test suite containing a test suite for each event. Actually, these are the most important tests to have since it's the main public API of this component (besides the properties), that would give something like:

```js
describe('events', function () {
    describe('`ez:cancel`', function () {
        it('should be dispatched');
    });

    describe('`ez:contentDiscovered`', function () {
        it('should be dispatched with a simple selection');

        it('should be dispatched with a multiple selection');
    });

    describe('`ez:select`', function () {
        it('should be dispatched when selecting an item');

        it('should allow to prevent the selection');
    });
});
```